fsgrid.hpp

```diff
@@ -26,99 +26,259 @@
 #include <limits>
 #include <stdint.h>
 #include <cassert>
-
+#include <stdio.h>
+#include <iomanip>
+#include <algorithm>
 
 
 
 struct FsGridTools{
 
+   typedef uint32_t FsSize_t; // Size type for global array indices
+   typedef int32_t FsIndex_t; // Size type for global/local array indices, incl. possible negative values
+
+   typedef int64_t LocalID;
```

Comment:
I dislike the use of LocalID and GlobalID here as we already use those in Vlasiator for velocity blocks.

Reply:
These are scoped to FsGridTools (or as it used to be, to plain FsGrid).